### PR TITLE
Add maximum segment number for every load segment request

### PR DESCRIPTION
**Pull request by sunby**
Signed-off-by: sunby <bingyi.sun@zilliz.com>
/kind improvement

### CODE REVIEW

File: internal/querycoordv2/task/merger.go [hunk 1]
```diff
@@ -128,12 +130,22 @@ func (merger *Merger[K, R]) mergeQueue(id K, queue chan MergeableTask[K, R]) {
 	defer merger.processors.Remove(id)
 
 	len := len(queue)
-	task := <-queue
-	for i := 1; i < len; i++ {
-		task.Merge(<-queue)
+	var task MergeableTask[K, R]
+	// output every per maxSegmentsPerTask
+	for i, curr := 1, 0; i < len; i, curr = i+1, (curr+1)%maxSegmentsPerTask {
```
Comment (yah01) [inline internal/querycoordv2/task/merger.go]: init `i = 1` but stop with condition `i < len`, which leaks one task

File: internal/querycoordv2/task/merger.go [hunk 2]
```diff
@@ -128,12 +130,22 @@ func (merger *Merger[K, R]) mergeQueue(id K, queue chan MergeableTask[K, R]) {
...
 	len := len(queue)
-	task := <-queue
-	for i := 1; i < len; i++ {
-		task.Merge(<-queue)
+	var task MergeableTask[K, R]
+	// output every per maxSegmentsPerTask
+	for i, curr := 0, 0; i < len; i, curr = i+1, (curr+1)%maxSegmentsPerTask {
+		if curr == 0 {
+			if task != nil {
+				merger.outCh <- task
```
Comment (yah01) [inline internal/querycoordv2/task/merger.go:138]: have to set `task = nil` after line 138, or the last task will be output again at line 146

**Review by yah01 — CHANGES_REQUESTED** [review]

**Review by congqixia — APPROVED** [review]
/lgtm

File: internal/querycoordv2/task/merger.go [hunk 3]
```diff
@@ -120,6 +120,8 @@ func (merger *Merger[K, R]) merge(id K, queue chan MergeableTask[K, R]) {
 	go merger.mergeQueue(id, queue)
 }
 
+const maxSegmentsPerTask = 3
```
Comment (jiaoew1991) [inline internal/querycoordv2/task/merger.go:123]: move it to configuration ?

Comment (yanliang567) [reply internal/querycoordv2/task/merger.go:123]: what does this impact? and what factors we need to think of if trying to update it

**Pull request closed without merging**